django_ibmi/client.py

```diff
@@ -32,27 +36,76 @@ class DatabaseClient(BaseDatabaseClient):
-        else:
-            cmdArgs += ["connect to %s" % database_name]
+        if 'PASSWORD' in settings_dict:
+            conn_params['password'] = settings_dict['PASSWORD']
+
+        if 'OPTIONS' in settings_dict:
+            conn_params.update(settings_dict['OPTIONS'])
+
+        allowed_opts = {'system', 'user', 'password', 'autocommit', 'readonly','timeout', 'database', 'use_system_naming',
+                        'library_list', 'current_schema'
+                        }
+
+        if not allowed_opts.issuperset(conn_params.keys()):
+            raise ValueError("Option entered not valid for "
+                             "IBM i Access ODBC Driver")
+
+        try:
+            conn_params['Naming'] = \
+                str(util.strtobool(conn_params['use_system_naming']))
+        except (ValueError, KeyError):
+            conn_params['Naming'] = '0'
 
-        if (isinstance(database_user, str) and
-                (database_user != '')):
-            cmdArgs += ["user %s" % database_user]
+        if 'current_schema' in conn_params or 'library_list' in conn_params:
+            conn_params['DefaultLibraries'] = \
+                conn_params.pop('current_schema', '') + ','
+            if isinstance(conn_params["DefaultLibraries"], str):
+                conn_params['DefaultLibraries'] += \
+                    conn_params.pop('library_list', '')
+            else:
+                conn_params['DefaultLibraries'] += ','.join(
+                    conn_params.pop('library_list', ''))
 
-            if (isinstance(database_password, str) and
-                    (database_password != '')):
-                cmdArgs += ["using %s" % database_password]
+        if os.name == 'nt':
+            cnxn = pyodbc.connect("Driver=IBM i Access ODBC Driver; UNICODESQL=1; TRUEAUTOCOMMIT=1;", **conn_params)
+            cursor = cnxn.cursor()
+            while True:
+                try:
+                    query = input("SQL> ")
+                    if query == "exit":
+                        break
+                    cursor.execute(query)
+                    try:
+                        print(cursor.fetchall())
+                    except pyodbc.ProgrammingError:
+                        pass
+                except ConnectionError as e:
+                    cursor.close()
+                    cnxn.close()
+                    print(e)
+                    break
+                except KeyboardInterrupt:
+                    cursor.close()
+                    cnxn.close()
+                    break
+            cursor.close()
+            cnxn.close()
 
-        # db2cmd is the shell which is required to run db2 commands on windows.
-        if (os.name == 'nt'):
-            os.execvp('db2cmd', cmdArgs)
         else:
-            os.execvp('db2', cmdArgs)
+            args = ['%s -v %s %s %s' %
+                    ('isql', settings_dict['NAME'], settings_dict['USER'], settings_dict['PASSWORD'])]
```

Comment:
Probably should use `-k` option to specify a connection string like we do elsewhere.